File: 이건희/level2/튜플.py
def solution(s):
    result = []
        
    s = s[:-2].replace('{','').replace(',',' ').split('}')
    s = [i.split() for i in s]
    s.sort(key=lambda x:len(x))
    
    for i in s:
        num = set(i) - set(result)
        result.append(list(num)[0])
    
    result = [int(i) for i in result]
    return result

File: 이훈일/Programmers/Level2/튜플.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        String sub = s.substring(2, s.length()-2);
        String rep = sub.replace("},{", " ");
        String[] spl = rep.split(" ");
        
        Arrays.sort(spl, (o1, o2) -> o1.length() - o2.length());
        
        List<Integer> tuple = new ArrayList<>();
        int[] result = new int[spl.length];
        for(int i = 0; i < spl.length; i++) {
            String[] numbers = spl[i].split(",");
            for(String number : numbers) {
                int iNumber = Integer.parseInt(number);
                if(!tuple.contains(iNumber)) {
                    tuple.add(iNumber);
                    result[i] = iNumber;
                }
            }
        }
        
        return result;
    }
}
